### PR TITLE
Modify S1774: Migrate to LayC - remove ternary expression

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S1774/cfamily/rule.adoc [hunk 1]
```diff
@@ -2,17 +2,26 @@
...
 
-=== Noncompliant code example
+=== Exceptions
+
+For {cpp}11 mode only, the issue is not raised for ternary operators used inside `constexpr` functions.
+In {cpp}11 such functions are limited to just a return statement, so the use of a ternary operator
+is required in them. This restriction is lifted in later standards, and thus issues are raised.
 
-[source,cpp]
+== How to fix it
```
Comment (frederic-tingaud-sonarsource) [inline rules/S1774/cfamily/rule.adoc:11]: The rule is very short and straightforward. I think having to change the tab to see the code sample reduces legibility.

Comment (rudy-regazzoni-sonarsource) [reply rules/S1774/cfamily/rule.adoc]: Mmhh, indeed that makes sense. I tried to simply remove the `How to fix it` section and to put examples directly in the `Why is this an issue?`, it seems to pass the test, I will go that way.

File: rules/S1774/cfamily/rule.adoc [hunk 2]
```diff
@@ -2,17 +2,24 @@
 
 include::../description.adoc[]
 
-=== Noncompliant code example
+=== Exceptions
```
Comment (frederic-tingaud-sonarsource) [inline rules/S1774/cfamily/rule.adoc:5]: Could you move this section after the code samples?

File: rules/S1774/cfamily/rule.adoc [hunk 3]
```diff
@@ -2,17 +2,24 @@
...
+For {cpp}11 mode only, the issue is not raised for ternary operators used inside `constexpr` functions.
+In {cpp}11 such functions are limited to just a return statement, so the use of a ternary operator
+is required in them. This restriction is lifted in later standards, and thus issues are raised.
 
-[source,cpp]
+=== Code examples
+
+==== Noncompliant code example
+
+[source,cpp,diff-id=1,diff-type=noncompliant]
```
Comment (frederic-tingaud-sonarsource) [inline rules/S1774/cfamily/rule.adoc:15]:
```suggestion
[source,cpp]
```
The whole block will be different, diff view doesn't really help in that case.

**Review by frederic-tingaud-sonarsource — CHANGES_REQUESTED** [review]

**Review by frederic-tingaud-sonarsource — APPROVED** [review]

**Pull request merged**